src/OpenLoco/Ui/Screenshot.cpp

```diff
@@ -117,6 +91,116 @@ namespace OpenLoco::Input
             png_destroy_write_struct(&pngPtr, nullptr);
             throw;
         }
+    }
+
+    // 0x00452667
+    static std::string prepareSaveScreenshot(Gfx::RenderTarget& rt)
+    {
+        auto basePath = Platform::getUserDirectory();
+        std::string scenarioName = S5::getOptions().scenarioName;
+
+        if (scenarioName.length() == 0)
+            scenarioName = StringManager::getString(StringIds::screenshot_filename_template);
+
+        std::string fileName = std::string(scenarioName) + ".png";
+        fs::path path;
+        int16_t suffix;
+        for (suffix = 1; suffix < std::numeric_limits<int16_t>().max(); suffix++)
+        {
+            if (!fs::exists(basePath / fileName))
+            {
+                path = basePath / fileName;
+                break;
+            }
+
+            fileName = std::string(scenarioName) + " (" + std::to_string(suffix) + ").png";
+        }
+
+        if (path.empty())
+        {
+            throw std::runtime_error("Failed finding filename");
+        }
+
+        std::fstream outputStream(path.c_str(), std::ios::out | std::ios::binary);
+        saveRenderTargetToPng(rt, outputStream);
+
+        return fileName;
+    }
+
+    std::string saveScreenshot()
+    {
+        auto& rt = Gfx::getScreenRT();
+        return prepareSaveScreenshot(rt);
+    }
+
+    std::string saveGiantScreenshot()
+    {
+        const auto& main = WindowManager::getMainWindow();
+        const auto zoomLevel = main->viewports[0]->zoom;
+
+        const uint16_t resolutionWidth = ((Map::kMapColumns * 32 * 2) >> zoomLevel) + 8;
+        const uint16_t resolutionHeight = ((Map::kMapRows * 32 * 1) >> zoomLevel) + 128;
+
+        Ui::Viewport viewport{};
+        viewport.width = resolutionWidth;
+        viewport.height = resolutionHeight;
+        viewport.x = 0;
+        viewport.y = 0;
+        viewport.viewWidth = viewport.width;
```

Comment:
No. This is wrong. It should be scaled by the zoomLevel. `ScreenToViewport::scaleTransform(viewDims, viewport)` would work but need to set the zoom first. Hmm might need to rework `ScreenToViewport` i didn't think it would be useful for stuff that isn't fully constructed.